### PR TITLE
Added tuning functionality for HDF5 output 

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

PR for testing. Discussion/review see #597 

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by pgrete** [comment]
@par-hermes format

**Review by Yurlungur — APPROVED** [review]

**Comment by Yurlungur** [comment]
looks like style check failed. @pgrete should we handle this? Or should we ask @brtnfld to fix the style? Should be fast to fix I think

**Comment by pgrete** [comment]
I went down the `type` rabbit hole as I suspected some potential issues with the lifetime of `const char*`.
Now separate string objects are used and passed with `c_str()` to the MPI function, which should guarantee they exist and don't contain garbage info when setting the parameters.
At the same time I also added some additional checks for boolean and `(h)size_t` environment parameters.
It'd be great if you,@Yurlungur and @brtnfld , could briefly review the last commit I made before we merge.

File: src/utils/utils.hpp
Comment (Yurlungur) [inline src/utils/utils.hpp:68]: Maybe add a comment that `parse_value` modifies the input string

**Review by Yurlungur — APPROVED** [review]
Your modifications seem fine to me, @pgrete . I approve. Maybe one thing is that the parse value methods in `Env` should be private? Or should be hidden in an `Impl` namespace?

File: src/utils/utils.hpp
Comment (pgrete) [reply src/utils/utils.hpp:68]: Both addressed.

**Comment by pgrete** [comment]
Technically, we need a second approval here. I cannot approve because I opened the PR, ping @jlippuner @jdolence @forrestglines

**Comment by brtnfld** [comment]
> I went down the `type` rabbit hole as I suspected some potential issues with the lifetime of `const char*`. Now separate string objects are used and passed with `c_str()` to the MPI function, which should guarantee they exist and don't contain garbage info when setting the parameters. At the same time I also added some additional checks for boolean and `(h)size_t` environment parameters. It'd be great if you,@Yurlungur and @brtnfld , could briefly review the last commit I made before we merge.

Looks good to me.

**Review by jlippuner — APPROVED** [review]
This looks good. Although I'm not sure why we're using environment variables here instead of the input file. Wouldn't it make more sense to have all these parameters specified in the input file like all the other Parthenon options? I guess downside is that we couldn't share input file between machines that need different tuning settings...

**Comment by pgrete** [comment]
Thanks! I'll go ahead an squash and merge then.

> This looks good. Although I'm not sure why we're using environment variables here instead of the input file. Wouldn't it make more sense to have all these parameters specified in the input file like all the other Parthenon options? I guess downside is that we couldn't share input file between machines that need different tuning settings...

With respect to the parameters, I actually think that these type of parameters are better suited for environment parameter (potentially included in a machine file as info or similar).
My reasoning is that those are independent from the simulation content itself and are tied to the compute "environment". Similarly, many other options and parameters set by scheduling/batch systems are environment variables so this seems like a good fit.

**Pull request merged**
